pkgs/development/tools/database/sqldeveloper/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, makeWrapper, requireFile, unzip, oraclejdk7}:
+
+stdenv.mkDerivation rec {
+  version = "4.0.3.16.84";
+  name = "sqldeveloper-${version}";
+
+  src = requireFile {
+    name = "${name}-no-jre.zip";
+    url = http://www.oracle.com/technetwork/developer-tools/sql-developer/downloads/index.html;
+    sha256 = "1qbqjkfda7xry716da2hdbbazks96rgyslrw1lw0azmqdp1mir7g";
+  };
+
+  buildInputs = [ makeWrapper unzip ];
+
+  buildCommand = ''
+    mkdir -p $out/bin
+    # patch to be able to install a sqldeveloper wrapper script compliant with nix's bin folder once installed
+    echo -e '#!/bin/bash\ncd "`dirname $0`"/../sqldeveloper/bin && bash sqldeveloper $*' >> $out/bin/sqldeveloper
+
+    cd $out
+    unzip ${src}
+    cp -r sqldeveloper/* $out/
+    # Activate the needed shell script
+    rm $out/sqldeveloper.sh
+    chmod +x $out/bin/sqldeveloper
+    chmod +x $out/sqldeveloper/bin/sqldeveloper
+
+    wrapProgram $out/bin/sqldeveloper \
+      --set JAVA_HOME "${oraclejdk7}"
```

Comment:
I remember now, it's because, it's sqldeveloper which claim to be in need of a jdk

Here is an extract of a configuration file (I do not use here because we set the JAVA_HOME) which speaks of JDK:

```txt
#
# By default, the product launcher will search for a JDK to use, and if none
# can be found, it will ask for the location of a JDK and store its location
# in this file. If a particular JDK should be used instead, uncomment the
# line below and set the path to your preferred JDK.
#
```